packages/core/android/src/main/java/io/sentry/react/RNSentryReactFragmentLifecycleTracer.java

```diff
@@ -95,6 +111,103 @@ public void onEventDispatch(Event event) {
             }
           }
         });
+
+    // Add layout listener to detect configuration changes after detaching any previous one
+    detachLayoutChangeListener();
+    attachLayoutChangeListener(v);
+  }
+
+  @Override
+  public void onFragmentViewDestroyed(@NonNull FragmentManager fm, @NonNull Fragment f) {
+    detachLayoutChangeListener();
+  }
+
+  private void attachLayoutChangeListener(final View view) {
+    final WeakReference<View> weakView = new WeakReference<>(view);
+
+    final ViewTreeObserver.OnGlobalLayoutListener listener =
+        new ViewTreeObserver.OnGlobalLayoutListener() {
+          @Override
+          public void onGlobalLayout() {
+            final View v = weakView.get();
+            if (v != null) {
+              checkAndNotifyWindowSizeChange(v);
+            }
+          }
+        };
+
+    currentView = view;
+    currentListener = listener;
+
+    view.getViewTreeObserver().addOnGlobalLayoutListener(listener);
+  }
+
+  private void detachLayoutChangeListener() {
+    if (currentView != null && currentListener != null) {
+      try {
+        ViewTreeObserver observer = currentView.getViewTreeObserver();
+        if (observer != null) {
+          observer.removeOnGlobalLayoutListener(currentListener);
+        }
+      } catch (Exception e) {
+        logger.log(SentryLevel.DEBUG, "Failed to remove layout change listener", e);
+      }
+    }
+
+    currentView = null;
+    currentListener = null;
+  }
+
+  private void checkAndNotifyWindowSizeChange(View view) {
+    try {
+      DisplayMetrics metrics = view.getContext().getResources().getDisplayMetrics();
+      int currentWidth = metrics.widthPixels;
+      int currentHeight = metrics.heightPixels;
+
+      if (lastWidth == currentWidth && lastHeight == currentHeight) {
+        return;
+      }
+      lastWidth = currentWidth;
+      lastHeight = currentHeight;
+
+      notifyReplayIntegrationOfSizeChange(currentWidth, currentHeight);
+    } catch (Exception e) {
+      logger.log(SentryLevel.DEBUG, "Failed to check window size", e);
+    }
+  }
+
+  private void notifyReplayIntegrationOfSizeChange(int width, int height) {
+    if (replayIntegration == null) {
+      replayIntegration = getReplayIntegration();
+    }
+
+    if (replayIntegration == null) {
+      return;
+    }
+
+    try {
+      replayIntegration.onWindowSizeChanged(width, height);
+    } catch (Exception e) {
+      logger.log(SentryLevel.DEBUG, "Failed to notify replay integration of size change", e);
+    }
+  }
+
+  private @Nullable ReplayIntegration getReplayIntegration() {
```

Comment:
nit: I think you could simplify this method by calling `ScopesAdapter.getInstance().getOptions().getReplayController()` and then checking if it's `instanceof ReplayIntegration`. This would spare a few cycles.